api_tests/invitations/by_emails.coffee

```diff
@@ -0,0 +1,172 @@
+      .catch undesiredErr(done)
+
+      return
+
+    it 'should accept several emails as an array', (done)->
+      authReq 'post', '/api/invitations?action=by-emails',
+        emails: [ 'a@foo.org', 'b@foo.org' ]
+      .then (res)->
+        res.emails[0].should.equal 'a@foo.org'
+        res.emails[1].should.equal 'b@foo.org'
+        done()
+      .catch undesiredErr(done)
+
+      return
+
+    it 'should reject missing emails', (done)->
+      authReq 'post', '/api/invitations?action=by-emails', {}
+      .then undesiredRes(done)
+      .catch (err)->
+        err.body.status_verbose.should.equal 'missing parameter in body: emails'
+        done()
+
+      return
+
+    it 'should reject invalid message', (done)->
+      authReq 'post', '/api/invitations?action=by-emails',
+        emails: 'a@foo.org'
+        message: []
+      .then undesiredRes(done)
+      .catch (err)->
+        err.statusCode.should.equal 400
+        err.body.status_verbose.should.match /invalid message:/
+        done()
+      .catch undesiredErr(done)
+
+      return
+
+    it 'should trigger an friend request on signup', (done)->
+      email = randomEmail()
+
+      invite = ->
+        authReq 'post', '/api/invitations?action=by-emails', { emails: email }
+
+      invite()
+      .then -> signup email
+      .then -> authReq 'get', '/api/relations'
+      .then (relations)->
+        invite()
+        .then (res)->
+          res.users[0].email.should.equal email
+          (res.users[0]._id in relations.userRequested).should.be.true()
+          done()
+      .catch undesiredErr(done)
+
+      return
+
+      email = randomEmail()
+      groupPromise
+      .then (group)->
+        authReq 'post', '/api/invitations?action=by-emails',
+          emails: email
+          group: group._id
+        .then -> signup email
+        .then -> getGroup group._id
+        .then (updatedGroup)->
+          prevInvitedCount = group.invited.length
+          invitedCount = updatedGroup.invited.length
+          invitedCount.should.equal prevInvitedCount + 1
+          done()
+
+      .catch undesiredErr(done)
+
+      return
+
+  describe 'groups', ->
+    it 'should reject invalid group ids', (done)->
+      authReq 'post', '/api/invitations?action=by-emails',
+        emails: 'a@foo.org'
+        group: 'abc'
+      .then undesiredRes(done)
+      .catch (err)->
+        err.statusCode.should.equal 400
+        err.body.status_verbose.should.equal 'invalid group id: abc'
+        done()
+      .catch undesiredErr(done)
+
+      return
+
+    it 'should accept valid group ids', (done)->
+      groupPromise
+      .then (group)->
+        authReq 'post', '/api/invitations?action=by-emails',
+          emails: 'a@foo.org'
+          group: group._id
+        .then (res)->
+          res.emails[0].should.equal 'a@foo.org'
+          done()
+      .catch undesiredErr(done)
+
+      return
+
+    it 'should reject non-user admin requests to invite to a group', (done)->
```

Comment:
I did it thinking that it was already the behavior for direct user invitations within a group, but actually everyone can invite their friends, so that limitation should be removed or placed behind an opt-in group setting. @pandark any opinion on this possibility?

Reply:
@jum-s meanwhile, I [converted the constraint into `reject non-member`](https://github.com/inventaire/inventaire/commit/fb0fd67ba6b08d8097fbdb0170129575c0bec965)